### DISCUSSION TITLE
Generals Pilot Mechanic

### DISCUSSION THREAD

**Discussion by ayylmaoRotE** [Phobos-developers/Phobos#883] [Ideas and feedback]
Add a tag which allows this infantry to enter a friendly vehicle (think reverse hijacker), then transfer veterancy over to that unit. If the unit is destroyed the pilot comes out with the unit's veterancy level.

If veterancy transfer is difficult, the pilot should then just add a single stage of veterancy, like a veteracy crate would.

e.g. IsPilot=yes

TLDR: Generals Pilot

**Comment by Metadorius** [comment]
How would it interact with already gained veterancy and units with a pilot already? Also is inf. piloting a neutral vehicle/hijacking an enemy vehicle an Ares feature?

**Comment by ayylmaoRotE** [comment]
If the vehicle already has veterancy, the pilot's veterancy will stack on top of it, up to heroic rank. 

Pilots can also enter vehicles on the player's same team, but the veterancy will be granted to the teammate instead.

If a hijacker takes over the vehicle, the pilot is ejected or killed, if this feature is implemented, pilots would not be able to hijack enemy vehicles unless they have the Hijack= tag too obviously, but that is up to you on how it would work. I vote for not being able to transfer veterancy. Neutral vehicles should be treated as friendly vehicles and gain veterancy based on the pilot's rank.
